lib/fastlane/actions/typetalk.rb

```diff
@@ -0,0 +1,47 @@
+module Fastlane
+  module Actions
+    class TypetalkAction
+      def self.run(params)
+        options = {
+            message: nil,
+            success: true,
+            topicId: nil,
+            typetalkToken: nil,
+        }.merge(params.first || {})
+
+        [:message, :topicId, :typetalkToken].each { |key|
+          raise "No #{key} given.".red unless options[key]
+        }
+
+        emoticon = (options[:success] ? ':smile:' : ':rage:')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.